hw6/hw6/ViewController.swift

```diff
@@ -0,0 +1,149 @@
+        
+        print(polyndrom("Too bad I hid a boot"))
+   }
+
+    func add(_ firstNumber: Int?, to secondNumber: Int?) -> String {
+        guard let firstNumber = firstNumber,
+              let secondNumber = secondNumber else {
+            return "Одно из значений не число"
+        }
+        return String(firstNumber + secondNumber)
+    }
+    
+    func multiply(_ firstNumber: Int?, _ secondNumber: Int?) -> String {
+        if let firstNumber = firstNumber,
+           let secondNumber = secondNumber {
+            return String(firstNumber * secondNumber)
+        }
+        return "Одно из значений не число"
+    }
+    
+    func myPow(_ firstNumber: Int?, _ secondNumber: Int?) -> String {
+        guard let firstNumber = firstNumber,
+              let secondNumber = secondNumber else {
+            return "Одно из значений не число"
+        }
+        return String(pow(Double(firstNumber), Double(secondNumber)))
+    }
+    
+    func subtraction(_ firstNumber: Int?, _ secondNumber: Int?) -> String {
+        guard let firstNumber = firstNumber,
+              let secondNumber = secondNumber else {
+            return "Одно из значений не число"
+        }
+        return String(firstNumber - secondNumber)
+    }
+    
+    func division(_ firstNumber: Double?, _ secondNumber: Double?) -> String {
+        guard let firstNumber = firstNumber,
+              let secondNumber = secondNumber else {
+            return "Одно из значений не число"
+        }
+        return String(firstNumber / secondNumber)
+    }
+    
+    func createCharacterArray(_ stringArray: [String?]) -> [Character] {
+        var characterArray = [Character]()
+
+        for element in stringArray {
+            guard let element = element else {
+                return ["E"]
+            }
+
+            for char in element {
+                characterArray.append(char)
+            }
+        }
+        return characterArray
+    }
+    
+    func countNumberRepeatedLettersOptionalStringArray(_ optionalStringArray: [String?]) {
+        let characterArray = createCharacterArray(optionalStringArray)
+        
+        let result = characterArray.reduce(into: [Character: Int]()) { partialResult, element in
+            partialResult[element, default: 0] += 1}
+        
+        print(result)
+    }
+    
+    func polyndrom(_ polyndromString: String?) -> String {
+        guard let polyndromString = polyndromString else {
+            return "Введите строку"
+        }
+        let lowerWithoutWhitespacesString = polyndromString.filter{!$0.isWhitespace}.lowercased()
+        var reverceString = String()
+        
+        for element in lowerWithoutWhitespacesString {
+            reverceString.insert(element, at: reverceString.startIndex)
+        }
+        
+        if lowerWithoutWhitespacesString == reverceString {
+            return "Строка палиндромом"
+        }
+        else {
+            return "Строка не палиндромом"
+        }
```

Comment:
Actually, you can drop `else` here because you have `return` inside `if` case so `return "Строка не палиндромом"` will only be called when the condition inside `if` isn't met